clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -306,7 +306,7 @@ public boolean isValid() {
     }
 
     private long computeChecksum() {
-        return Crc32.crc32(buffer, ATTRIBUTES_OFFSET, buffer.limit() - ATTRIBUTES_OFFSET);
+        return Crc32C.compute(buffer, ATTRIBUTES_OFFSET, buffer.limit() - ATTRIBUTES_OFFSET);
```

Comment:
Maybe we can add the fact that we use crc32c in the javadoc to this class which describes the batch schema.